### PR TITLE
[#1555] added a "fix" for the question group bug

### PR DESCRIPTION
**Pull request by valllllll2000**
#### Before the PR (what is the issue or what needed to be done)
If a repeated question was left unanswered and you left the datapoint and opened the saved datapoint again, the first repetition was fine but after the second one, the responses were filled on the empty questions using the ones for the first repetition.
#### The solution
* repeated group responses will no longer be duplicated on empty questions
* added comments to explain and a TODO
#### Screenshots (if appropriate)
NA
#### Reviewer Checklist
* [x] Added an explanation about the work done
* [x] Connected the PR and the issue on Zenhub
* [ ] Added a test plan to the issue
* [ ] Updated the copyright header
* [ ] Formatted the code per our style guide
* [ ] Added a documentation (if relevant)

### CODE REVIEW

File: app/src/main/java/org/akvo/flow/ui/view/QuestionGroupTab.java [hunk 1]
```diff
@@ -201,15 +201,31 @@ private void updateGroupIterationHeaders() {
         }
     }
 
+    //TODO: this method should be refactored to find a better way to distinguish between
```
Comment (muloem) [inline app/src/main/java/org/akvo/flow/ui/view/QuestionGroupTab.java]: Create a separate issue for this and then we can clear out this comment

Comment (valllllll2000) [inline app/src/main/java/org/akvo/flow/ui/view/QuestionGroupTab.java]:
```suggestion
```

File: app/src/main/java/org/akvo/flow/ui/view/QuestionGroupTab.java [hunk 2]
```diff
@@ -201,15 +201,30 @@ private void updateGroupIterationHeaders() {
         }
     }
 
+    //repeatable and non repeatable groups
```
Comment (valllllll2000) [inline app/src/main/java/org/akvo/flow/ui/view/QuestionGroupTab.java]:
```suggestion
```

File: app/src/main/java/org/akvo/flow/ui/view/QuestionGroupTab.java (hunk 1 above)
Comment (valllllll2000) [reply app/src/main/java/org/akvo/flow/ui/view/QuestionGroupTab.java]: Done https://github.com/akvo/akvo-flow-mobile/issues/1557

**Pull request merged**